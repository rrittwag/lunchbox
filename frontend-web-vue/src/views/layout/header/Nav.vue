<template>
  <nav class="flex items-center justify-center sm:h-16" aria-label="Haupt">
    <ul class="list-none">
      <slot></slot>
    </ul>
  </nav>
</template>
