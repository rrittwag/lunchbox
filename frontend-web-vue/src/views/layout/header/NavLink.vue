<script setup lang="ts">
import { RouterLink } from 'vue-router'

const props = defineProps<{
  to: string
}>()
</script>

<template>
  <li class="inline-block px-2">
    <RouterLink v-slot="{ navigate, isExactActive, route }" :to="props.to" custom>
      <a
        class="nav-link flex h-12 w-12 items-center justify-center"
        :class="isExactActive ? 'text-neutral-800' : 'text-neutral-300'"
        :href="route.path"
        :title="route.meta.title"
        :aria-label="route.meta.title"
        :aria-current="isExactActive ? 'page' : undefined"
        @click="navigate"
      >
        <slot />
      </a>
    </RouterLink>
  </li>
</template>

<style lang="scss">
.nav-link > svg {
  @apply h-8 w-8;
}
</style>
