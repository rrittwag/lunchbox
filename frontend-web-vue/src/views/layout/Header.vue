<template>
  <header class="pb-2 pt-4 sm:absolute sm:right-0 sm:top-0 sm:px-8 sm:pb-4">
    <Nav>
      <NavLink to="/">
        <OffersIcon aria-hidden="true" />
      </NavLink>

      <NavLink to="/settings">
        <SettingsIcon aria-hidden="true" />
      </NavLink>

      <NavLink to="/about">
        <InfoIcon aria-hidden="true" />
      </NavLink>
    </Nav>
  </header>
</template>

<script setup lang="ts">
import OffersIcon from '~icons/fa-solid/utensils'
import SettingsIcon from '~icons/fa-solid/cog'
import InfoIcon from '~icons/fa-solid/info-circle'
import NavLink from '@/views/layout/header/NavLink.vue'
import Nav from '@/views/layout/header/Nav.vue'
</script>
