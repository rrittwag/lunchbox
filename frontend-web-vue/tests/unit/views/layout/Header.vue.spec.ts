import Header from '@/views/layout/Header.vue'
import NavLink from '@/views/layout/header/NavLink.vue'
import Nav from '@/views/layout/header/Nav.vue'
import { mount } from '@vue/test-utils'
import { createRouterMock, injectRouterMock } from 'vue-router-mock'

describe('Header', () => {
  const router = createRouterMock({})
  beforeEach(() => {
    injectRouterMock(router)
  })

  test('renders snapshot', () => {
    const wrapper = mount(Header)

    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders header tag and Nav component', () => {
    const wrapper = mount(Header)

    const headerTag = wrapper.get('header')
    const navTag = headerTag.getComponent(Nav)
    expect(navTag.findAllComponents(NavLink)).toHaveLength(3)
  })

  test('has all nav items', () => {
    const wrapper = mount(Header)

    const navitems = wrapper.findAllComponents(NavLink)
    expect(navitems[0].props('to')).toEqual('/')
    expect(navitems[1].props('to')).toEqual('/settings')
    expect(navitems[2].props('to')).toEqual('/about')
  })
})
