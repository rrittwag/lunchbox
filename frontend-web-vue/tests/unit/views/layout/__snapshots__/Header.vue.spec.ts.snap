// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders snapshot 1`] = `
<header
  class="pt-4 pb-2 sm:absolute sm:top-0 sm:right-0 sm:px-8 sm:pb-4"
>
  <nav
    aria-label="Haupt"
    class="flex items-center justify-center sm:h-16"
  >
    <ul
      class="list-none"
    >
      
      <li
        class="inline-block px-2"
      >
        <router-link-stub
          custom=""
          to="/"
        />
      </li>
      <li
        class="inline-block px-2"
      >
        <router-link-stub
          custom=""
          to="/settings"
        />
      </li>
      <li
        class="inline-block px-2"
      >
        <router-link-stub
          custom=""
          to="/about"
        />
      </li>
      
    </ul>
  </nav>
</header>
`;
